### PR TITLE
feat(storage): reduce group deltas for l0 task

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by hzxa21 — APPROVED** [review]
LGTM. But I wonder how much it can save. IIUC, it only saves the pb overhead?

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Comment by Little-Wallace** [comment]
> LGTM. But I wonder how much it can save. IIUC, it only saves the pb overhead?

We want to store group-deltas of several compact-task in one version-delta, so we shall simplify the code of version update. I think this PR would not save much CPU or latency by itself.

File: src/meta/src/hummock/manager/mod.rs [hunk 1]
```diff
@@ -2775,34 +2776,52 @@ fn gen_version_delta<'a>(
...
+        match removed_table_ids_map.entry(level_idx) {
+            Vacant(entry) => {
+                entry.insert(removed_table_ids);
+            }
+
+            Occupied(mut entry) => {
+                entry.get_mut().append(&mut removed_table_ids);
+            }
+        };
+    }
```
[the remark this answers is not in the document]
Comment (Li0k) [reply src/meta/src/hummock/manager/mod.rs:2812]: Indeed, it's a minor change .

It only reduces the number of group deltas in a single pb, for example:
The l0 task contains 42 levels of sublevels, and the pr reduces the number of group deltas from 42(select level) + 1(target level) + 1 (inserted) -> 1(select level) + 1(target level) + 1 (inserted).

**Pull request merged**